enhancements/machine-config/pin-and-pre-load-images.md

```diff
@@ -164,71 +204,60 @@ pinned_images=[
-different sets of images for different kinds of nodes. For example, to pin
-different images for control plane and worker nodes the user could create two
-`PinnedImageSet` custom resources:
-
-```yaml
-# For control plane nodes:
-apiVersion: machineconfiguration.openshift.io/v1alpha1
-kind: PinnedImageSet
-metadata:
-  name: my-control-plane-pinned-images
-spec:
-  nodeSelector:
-    matchLabels:
-      node-role.kubernetes.io/control-plane: ""
-  pinnedImages:
-  ...
-
----
-
-# For worker nodes:
-apiVersion: machineconfiguration.openshift.io/v1alpha1
-kind: PinnedImageSet
-metadata:
-  name: my-worker-pinned-images
-spec:
-  nodeSelector:
-    matchLabels:
-      node-role.kubernetes.io/worker: ""
-  pinnedImages:
-  ...
-```
+The PinnedImageSet is closely linked with the `MachineConfigPool`, and each
+Custom Resource (CR) can be associated with a pool at the
+`MachineConfigPoolSpec` level. This design is advantageous because it enables
+controllers or administrators to assign specific sets of images to different
+pools. By default, a cluster has two types of pools: master and worker. However,
+users can create [custom pools](https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md)
+for more precise control.
 
 This is specially convenient for pinning images for upgrades: there are many
 images that are needed only by the control plane nodes and there is no need to
 have them consuming disk space in worker nodes.
 
-When no node selector is specified the images will be pinned in all the nodes
-of the cluster.
-
-The _machine-config-controller_ will grow a new `PinnedImageSetController` sub
-controller that will watch the `PinnedImageSet` custom resources. When one of
-those is created, updated or deleted the controller will start a daemon set that
-will do the following in each node of the cluster:
+The _machine-config-daemon_ will grow a new `pinned_image_manager` utilizing
+the same general flow as the existing `MAchineConfigDaemon` `certificate_writer`. This approach is not dependent on
+defining the configuration as `MachineConfig`. The controller that will watch on
+`PinnedImageSet` and `MachineConfigPool` resources. On sync the manager will
+perform the following tasks.
+
+The _machine-config-daemon_ will add the following metrics for admins to track/verify image prefetch progress.
+
+- mcd_prefetch_image_pull_success_total: Total number of successful prefetched image pulls.
+
+- mcd_prefetch_image_pull_failure_total: Total number of prefetch image pull failures.
+
+The _machine-config-daemon_ will be enhanced with a new feature called
+`pinned_image_manager`, which follows a similar process to the existing
+`certificate_writer`. This new feature will operate independently of the
+`MachineConfig` for setting configurations. It is advantageous to not use
+`MachineConfig` because the configuration and image prefetching can happen
+across the nodes in the pool in parallel. The new controller will be watching
```

Comment:
Now that we are introducing `PinnedImageSetController` , which `new controller` we are referring to here?

Reply:
sorry, it was a typo. I meant to ask, `Now that we are no longer adding PinnedImageSetController , which new controller we are referring to here?